numpy/core/src/multiarray/scalarapi.c

```diff
@@ -486,8 +483,8 @@ PyArray_DescrFromTypeObject(PyObject *type)
             new->elsize = conv->elsize;
             new->subarray = conv->subarray;
             conv->subarray = NULL;
-            Py_DECREF(conv);
         }
+        Py_DECREF(conv);
```

Comment:
This is probably quite a good argument for `INCREF`ing `Py_NotImplemented` before returning it.

Reply:
:+1: Not incrementing is just too smart anyway...